File: Dynamic Programming/numberOfWaysToMakeChange.js
const numberOfWaysToMakeChange = (n, denoms) => {
	let ways = new Array(n + 1).fill(0)
	ways[0] = 1
	
	for (const denom of denoms) {
        let amount = 1
		while (amount <= n) {
			if (denom <= amount) {
			  ways[amount] += ways [amount - denom]
            }
            amount++
		}
	}
	return ways[n]
}

File: Dynamic Programming/number_of_ways_to_make_change.py
def number_of_ways_to_make_change(target, denoms):
    ways = [0] * (target + 1)
    ways[0] = 1

    for denom in denoms:
        for amount in range(1, target + 1):
            if (denom <= amount):
                ways[amount] += ways[amount - denom]
    return ways[target]